linera-core/src/unit_tests/client_tests.rs

```diff
@@ -1488,6 +1488,130 @@ where
     Ok(())
 }
 
+#[test_case(MemoryStorageBuilder::default(); "memory")]
+#[cfg_attr(feature = "storage_service", test_case(ServiceStorageBuilder::new().await; "storage_service"))]
+#[cfg_attr(feature = "rocksdb", test_case(RocksDbStorageBuilder::new().await; "rocks_db"))]
+#[cfg_attr(feature = "dynamodb", test_case(DynamoDbStorageBuilder::default(); "dynamo_db"))]
+#[cfg_attr(feature = "scylladb", test_case(ScyllaDbStorageBuilder::default(); "scylla_db"))]
+#[test_log::test(tokio::test)]
+async fn test_read_blob1<B>(storage_builder: B) -> anyhow::Result<()>
+where
+    B: StorageBuilder,
+    ViewError: From<<B::Storage as Storage>::StoreError>,
+{
+    // Configure a chain with two regular and no super owners.
+    let mut builder = TestBuilder::new(storage_builder, 4, 0).await?;
+    let description1 = ChainDescription::Root(1);
+    let description2 = ChainDescription::Root(2);
+    let chain_id1 = ChainId::from(description1);
+    let mut client1_a = builder
+        .add_initial_chain(description1, Amount::from_tokens(10))
+        .await?;
+    let mut client2_a = builder
+        .add_initial_chain(description2, Amount::ZERO)
+        .await?;
+    let pub_key0 = client1_a.public_key().await.unwrap();
+    let key_pair1 = KeyPair::generate();
+    let pub_key1 = key_pair1.public();
+    let owner_change_op = SystemOperation::ChangeOwnership {
+        super_owners: Vec::new(),
+        owners: vec![(pub_key0, 100), (pub_key1, 100)],
+        multi_leader_rounds: 10,
+        timeout_config: TimeoutConfig {
+            fast_round_duration: Some(TimeDelta::from_secs(5)),
+            ..TimeoutConfig::default()
+        },
+    }
+    .into();
+    client1_a.execute_operation(owner_change_op).await.unwrap();
+    let mut client1_b = builder
+        .make_client(
+            chain_id1,
+            key_pair1,
+            client1_a.block_hash(),
+            BlockHeight::from(1),
+        )
+        .await?;
+
+    let blob = HashedBlob::test_blob("blob1");
+    let expected_blob_id = blob.id();
+    // Try to read a blob without publishing it first, should fail
+    let result = client1_a.read_blob(expected_blob_id).await;
+    assert_matches!(
+        result,
+        Err(ChainClientError::BlobNotFound(blob_id)) if blob_id == expected_blob_id
+    );
+
+    // Take one validator down
+    builder.set_fault_type([3], FaultType::Offline).await;
+
+    // Publish blob
+    let (blob_id, certificate) = client1_a.publish_blob(blob).await.unwrap().unwrap();
+    client1_a.synchronize_from_validators().await.unwrap();
+    assert_eq!(expected_blob_id, blob_id);
+    assert_eq!(certificate.round, Round::MultiLeader(0));
+    assert!(certificate
+        .value()
+        .executed_block()
+        .unwrap()
+        .outcome
+        .oracle_records
+        .iter()
+        .any(|record| record
+            .responses
+            .contains(&OracleResponse::Blob(expected_blob_id))));
```

Comment:
Doesn't `ExecutedBlock` have a method for that?

Reply:
Not for this specifically, but good point, I'll add it